src/main.rs

```diff
@@ -102,16 +102,37 @@ fn real_main(
     } else {
         &opt.opt
     };
-    let (result, warnings) = compile(buf, &opt);
-    handle_warnings(warnings, file_id, file_db);
+    if !opt.jit {
+        let (result, warnings) = compile_aot(buf, &opt);
+        handle_warnings(warnings, file_id, file_db);
 
-    let product = result?;
-    if opt.no_link {
-        return assemble(product, output);
+        let product = result?;
+        if opt.no_link {
+            return assemble(product, output);
+        }
+        let tmp_file = NamedTempFile::new()?;
+        assemble(product, tmp_file.as_ref())?;
+        link(tmp_file.as_ref(), output).map_err(io::Error::into)
+    } else {
+        let (jit, warnings) = compile_jit(buf, &opt);
+        handle_warnings(warnings, file_id, file_db);
+        let mut jit = jit?;
+        jit.finalize();
+        let main_function = jit.get_compiled_function("main");
+        if let Some(main_function) = main_function {
+            let args = std::env::args();
+            let argc = args.len() as i32;
+            let vec_args = args
+                .into_iter()
+                .map(|string| std::ffi::CString::new(string).unwrap().as_ptr() as *const u8)
+                .collect::<Vec<*const u8>>();
+            let pointer = vec_args.as_ptr() as *const *const u8;
+            let main: unsafe extern "C" fn(i32, *const *const u8) -> i32 =
+                unsafe { std::mem::transmute(main_function) };
+            let _ = unsafe { main(argc, pointer) };
```

Comment:
Maybe it would be better to have a separate binary for JIT? I'm not sure I like the idea of RCC crashing at runtime if there's bad codegen.

Reply:
> I have question there: what we should do with main return value? Should we pass it to std::process::exit? Currently this value just ignored

Passing to exit() is better than before but I'm still not really happy with it. I would rather this be a library function somewhere that e.g. a REPL could call to get the return code.